OpenSim/Region/OptionalModules/Avatar/FlexiGroups/FlexiGroupsModule.cs

```diff
@@ -1034,14 +1034,29 @@ public UUID CreateGroup(IClientAPI remoteClient, string name, string charter, bo
                 remoteClient.SendCreateGroupReply(UUID.Zero, false, "A group with the same name already exists.");
                 return UUID.Zero;
             }
-            
-            UUID groupID = m_groupData.CreateGroup(grID, name, charter, showInList, insigniaID, membershipFee, openEnrollment, allowPublish, maturePublish, remoteClient.AgentId);
 
-            remoteClient.SendCreateGroupReply(groupID, true, "Group created successfullly");
+            IScene scene = (remoteClient == null) ? m_sceneList[0] : remoteClient.Scene;
+            IMoneyModule mm = scene.RequestModuleInterface<IMoneyModule>();
+            if (mm == null)
+            {
+                remoteClient.SendAgentAlertMessage("Server configuration error - cannot create group without money module.", false);
+                return UUID.Zero;
+            }
+
+            if (!mm.GroupCreationCovered(remoteClient.AgentId)) {
+                remoteClient.SendAgentAlertMessage("Unable to create group. Insufficient funds.", false);
+                return UUID.Zero;
+            }
+
+            UUID groupID = m_groupData.CreateGroup(grID, name, charter, showInList, insigniaID, membershipFee, openEnrollment, allowPublish, maturePublish, remoteClient.AgentId);
+            if (groupID != UUID.Zero)
+            {
+                mm.ApplyGroupCreationCharge(remoteClient.AgentId);
+                remoteClient.SendCreateGroupReply(groupID, true, "Group created successfullly");
+            }
```

Comment:
No error to the user if the group failed to be created?

Reply:
Good point. I improved the previous behavior, which was to always report the group creation as a success, even when it failed, by making that result conditional on actual success. I didn't add a new message for the failure case, and maybe that would be a good improvement here.